### ISSUE TITLE
Extracted submechanism causes failure with CounterflowDiffusionFlame

### ISSUE THREAD

**Issue by d-e-t** [Cantera/cantera#1067]
**Problem description**

A submechanism created using the extract_submechanism.py example causes a `CounterflowDiffusionFlame` simulation to fail, returning only a `floating point exception`, and no further debug info (even with `loglevel=5`).  The submechanism does not cause failure with `CounterflowPremixedFlame` or `IdealGasConstPressureReactor`.  I also see the same behavior  when using mechanisms other than gri30.yaml

**Steps to reproduce**

Run the attached modified version of extract_submechanism.py.  The simulation runs as normal with the full mechanism, but fails with the submechanism.  Uncomment the original `CounterflowPremixedFlame` function for a solution with both the full mechanism and submechanism.

**System information**

- Cantera: same results with 2.5.1 or dev version 2.6.0a2
- OS: Mac OS 11.4 and Windows 10
- Python 3.9

**Attachments**

[extract_submechanism_bug.zip](https://github.com/Cantera/cantera/files/6748698/extract_submechanism_bug.zip)

**Additional context**

Looking at the extracted submechanism after writing with write_yaml, I don't see any obvious errors.  I have also tried removing undeclared third-body species from the submechanism, but without any change in behavior.

**Comment by d-e-t** [comment]
[mechanism_reduction_bug.py.zip](https://github.com/Cantera/cantera/files/6773029/mechanism_reduction_bug.py.zip) is another example showing the same behavior with a modified version of the mechanism_reduction.py example.  Again, the `CounterflowPremixedFlame` works fine with the reduced mechanism, but `CounterflowDiffusionFlame` fails with a `floating point exception` for any reduced mechanism -- even if only 1 reaction is removed.

If anyone has suggestions for me on directions to go with digging into this, or ways to get at other debug/diagnostic, I'd be very grateful.

**Comment by d-e-t** [comment]
The problem is related to initial state of the phase.  In both the examples I posted, the `CounterflowDiffusionFlame` runs fine if the reduced/sub-mechanism state is initialized before passing into the new flame object: 
`gas2.TP = 300, ct.one_atm` 
This is of course a simple work-around, but it seems like the problem should still be fixed -- perhaps in the initialization of the `Solution` or `CounterflowDiffusionFlame`?  Is the initial state of the `CounterflowDiffusionFlame` handled differently from other flames?  Or the problem could be resolved by a different default state when Solution is created using a set of species and reactions instead of a mechanism file?  

When initializing a Solution object with
`gas2 = ct.Solution(thermo='ideal-gas', kinetics='gas', species=species_list, reactions=reaction_list)`
The default gas2.TP is: 0.001, 0.0041

**Comment by speth** [comment]
Thanks for reporting the `floating point exception` errors. Even with problematic input, the 1D solver should at worst only ever fail with a `CanteraError`. I traced this back to an issue with the error handling code in `MultiNewton::step`. This should be fixed by #1103. Now, the error reported is
```
CanteraError thrown by MultiNewton::step:
Jacobian is singular for domain flame, component lambda at point 5
(Matrix row 88)
```
which is not really that much more useful, but at least it isn't crashing outright.

As you've identified, this does come down to trying to solve the flame at an unrealistically low pressure (0.0041 Pa). Unlike the `gri30.yaml` file, which specifies an initial temperature and pressure, creating a phase by just giving the phase model and a list of species leaves the phase with the default state, which is for temperature and density to be set to 0.001 and a mass fraction of 1 for the first species. This is done just so phase setup can be done without generating a bunch of `NaN`s. While we could use a different default state, I feel like this would just lead to more undetected errors from using a `Solution` object without correctly setting its state.

**Comment by d-e-t** [comment]
Nice!  Thanks for tracking that down, Ray.

Yes, keeping the default state as is makes sense to me.  Looking back at my
example code, I see I had not set the pressure of the Solution object for
the CounterflowDiffusionFlame; initializing the pressure there takes care
of any differences between flame objects.  So I'd say this issue could be
closed.

**Comment by speth** [comment]
With the second change introduced in #1103, the initial condition for the flame solver is now good enough to actually be able to solve the case demonstrated here. So with that and your suggestion above, I think we can let this be closed by #1103.
